File: src/main/scala/lang/scala/core/NestedClass.scala
package lang.scala.core

import NestedClass.Car.Tire

object NestedClass extends App {
  class Car(val year: Int)
  object Car {
    /**
     * Nested class
     */
    class Tire(val radius: Double)
  }

  val car = new Car(2015)
  val tire = new Tire(15.0)
  println(tire.radius)
}

File: src/main/java/lang/java/core/NestedClass.java
package lang.java.core;

class Car {
    private int year;

    /**
     * Nested class
     */
    public static class Tire {
        private double radius;

        public Tire(double radius) {
            this.radius = radius;
        }

        public double getRadius() {
            return radius;
        }
    }

    public Car(int year) {
        this.year = year;
    }

    public int getYear() {
        return year;
    }
}

public class NestedClass {
   public static void main(String[] args) {
       Car car = new Car(2015);
       Car.Tire tire = new Car.Tire(15.0); // nested class init
       System.out.println(tire.getRadius());
   }
}
